src/codegen/encoding/mod.rs

```diff
@@ -428,39 +418,28 @@ fn calculate_struct_size<T: AbiEncoding>(
     size
 }
 
-/// Loads an item from an array
-fn load_array_item(arr: &Expression, dims: &[ArrayLength], indexes: &[usize]) -> Expression {
-    let elem_ty = arr.ty().elem_ty();
-    let (deref, ty) = load_sub_array(arr.clone(), dims, indexes, false);
-    Expression::Subscript(
-        Loc::Codegen,
-        Type::Ref(Box::new(elem_ty)),
-        ty,
-        Box::new(deref),
-        Box::new(Expression::Variable(
-            Loc::Codegen,
-            Type::Uint(32),
-            *indexes.last().unwrap(),
-        )),
-    )
-}
-
-/// Dereferences a subarray. If we have 'int[3][][4] vec' and we need 'int[3][]',
-/// this function returns so.
-/// 'dims' should contain only the dimensions we want to index
-/// 'index' is the list of indexes to use
-/// 'index_first_dim' chooses whether to index the first dimension in dims
-fn load_sub_array(
+/// Indexes an array. If we have 'int[3][][4] vec' and we need 'int[3][]',
+/// 'int[3]' or 'int' this function returns so.
+/// 'dims' is the vector containing the arrays dimensions
+/// 'index' is the list of indexes to use for each dimension
+/// 'return_pointer' tells us if we should return a pointer
```

Comment:
This comment is not helpful, as that is already obvious by looking at the code. It would be better to explain why and/or when a user would want that.

Reply:
I'll improve the comment.

> as it is already obvious what return_pointer does by looking at the code.

Well, if you analyze well, you'll see that even if `return_pointer` is false, you can have a pointer returned, so it is not as obvious as you said.

Reply:
I've updated all the comments you pointed. Can you please re-check?

Reply:
Looks good, thanks